bin/stacks/api-stack.ts

```diff
@@ -342,23 +342,115 @@ export class APIStack extends cdk.Stack {
+      alarmName: 'UnifiedRoutingAPI-SEV2-LatencyP99WithDeps',
+      actionsEnabled: true,
+      evaluationPeriods: 3,
+      datapointsToAlarm: 3,
+      threshold: 1,
+      comparisonOperator: aws_cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric:
+        new aws_cloudwatch.MathExpression({
+          expression: "IF(ura_high_latency AND low_routing_api_latency, 1, 0)",
+          label: 'Latency Alarm',
+          usingMetrics: {
+            ura_high_latency: new aws_cloudwatch.MathExpression({
+              expression: "IF(overall_latency > 10000, 1, 0)",
+              label: 'Overall Latency',
+              usingMetrics: {
+                overall_latency: new aws_cloudwatch.Metric({
+                  namespace: 'AWS/ApiGateway',
+                  metricName: 'Latency',
+                  dimensionsMap: {
+                    ApiName: 'UnifiedRouting'
+                  },
+                  statistic: 'p99',
+                }),
+              },
+            }),
+            low_routing_api_latency: new aws_cloudwatch.MathExpression({
+              expression: "IF(routing_api_latency < 4000, 1, 0)",
+              label: 'Routing API Quoter Latency',
+              usingMetrics: {
+                routing_api_latency: new aws_cloudwatch.Metric({
+                  namespace: 'Uniswap',
+                  metricName: 'RoutingApiQuoterLatency',
+                  dimensionsMap: {
+                    Service: 'UnifiedRoutingAPI'
+                  },
+                  statistic: 'p99',
+                }),
+              },
+            }),
+          },
+        }),
+    });
+
     const apiAlarmLatencyP99Sev3 = new aws_cloudwatch.Alarm(this, 'UnifiedRoutingAPI-SEV3-LatencyP99', {
       alarmName: 'UnifiedRoutingAPI-SEV3-LatencyP99',
       metric: api.metricLatency({
-        period: Duration.minutes(5),
+        period: Duration.minutes(20),
         statistic: 'p99',
       }),
       threshold: 7000,
       evaluationPeriods: 3,
     });
 
+    // Alarm if URA latency is high (> 7s) and Routing API is not (< 4s)
+    // Usually there's nothing to be done in URA when RoutingAPI latency is high
+    const apiAlarmLatencyP99WithDepsSev3 = new aws_cloudwatch.Alarm(this, 'UnifiedRoutingAPI-SEV3-LatencyP99WithDeps', {
+      alarmName: 'UnifiedRoutingAPI-SEV3-LatencyP99WithDeps',
+      actionsEnabled: true,
+      evaluationPeriods: 3,
+      datapointsToAlarm: 3,
+      threshold: 1,
+      comparisonOperator: aws_cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+      metric:
+        new aws_cloudwatch.MathExpression({
+          expression: "IF(ura_high_latency AND low_routing_api_latency, 1, 0)",
+          label: 'Latency Alarm',
+          usingMetrics: {
+            ura_high_latency: new aws_cloudwatch.MathExpression({
+              expression: "IF(overall_latency > 7000, 1, 0)",
```

Comment:
nit: would be nice to put these numbers into constants (URA_LATENCY_SEV3 = 7000, ROUTING_API_LATENCY_SEV3 = 4000, etc.) so we can change them all in one place

Reply:
Great callout! Moved these magic numbers to the constants file.